go/pkg/tree/tree.go

```diff
@@ -48,7 +48,7 @@ func shouldIgnore(inp string, t command.InputType, excl []*command.InputExclusio
 // loadFiles reads all files specified by the given InputSpec (descending into subdirectories
 // recursively), and loads their contents into the provided map.
 func loadFiles(execRoot string, excl []*command.InputExclusion, path string, fs map[string]*chunker.Chunker, chunkSize int, cache FileMetadataCache) error {
-	absPath := filepath.Join(execRoot, path)
+	absPath := filepath.Clean(filepath.Join(execRoot, path))
 	meta, err := cache.Get(absPath)
 	t := command.FileInputType
 	if err != nil {
```

Comment:
The check below seems brittle. Depending on the FileMetadataCache implementation, the error message could be wrapped/modified. I suggest using IsDir to check if the input is a directory.

Reply:
Discussed offline, added the FileError type.